File: frontend/lib/models/user_profile.dart
class UserProfile {
  final String uid;
  final String phoneNumber;
  final String? displayName;
  final String createdAt;

  UserProfile({
    required this.uid,
    required this.phoneNumber,
    this.displayName,
    required this.createdAt,
  });

  factory UserProfile.fromJson(Map<String, dynamic> json) {
    return UserProfile(
      uid: json['uid'] as String? ?? '',
      phoneNumber: json['phone_number'] as String? ?? '',
      displayName: json['display_name'] as String?,
      createdAt: json['created_at'] as String? ?? '',
    );
  }
}

File: backend/app/schemas/user_profile.py
"""Schemas for user profile (phone number storage after Google Auth)."""

from pydantic import BaseModel


class UserProfileInput(BaseModel):
    """Input body for PUT /api/v1/profile."""
    phone_number: str          # "+923001234567"
    display_name: str | None = None


class UserProfileResponse(BaseModel):
    """Response from GET/PUT /api/v1/profile."""
    uid: str
    phone_number: str
    display_name: str | None = None
    created_at: str
